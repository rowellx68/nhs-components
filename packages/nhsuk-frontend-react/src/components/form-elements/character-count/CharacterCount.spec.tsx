import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  WithMaxCharacters as WithMaxCharactersStory,
  WithMaxWords as WithMaxWordsStory,
} from './CharacterCount.stories';

const WithMaxCharacters = composeStory(WithMaxCharactersStory, meta);
const WithMaxWords = composeStory(WithMaxWordsStory, meta);

it('should render the CharacterCount component with max characters', () => {
  const { container } = render(<WithMaxCharacters />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the CharacterCount component with max words', () => {
  const { container } = render(<WithMaxWords />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
