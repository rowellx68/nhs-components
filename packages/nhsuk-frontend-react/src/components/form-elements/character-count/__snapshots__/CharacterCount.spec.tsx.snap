// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the CharacterCount component with max characters 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-maxlength="10"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r0:"
        id="textarea_:r0:--label"
      >
        Can you provide more detail?
      </label>
      <div
        class="nhsuk-hint"
        id="textarea_:r0:_:r1:--hint"
      >
        Do not include personal or financial information, for example, your National Insurance number or credit card details.
      </div>
      <textarea
        aria-describedby="textarea_:r0:_:r1:--hint"
        aria-labelledby="textarea_:r0:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="textarea_:r0:"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message nhsuk-u-visually-hidden"
        id="textarea_:r0:-info"
      >
        You have 10 characters remaining
      </div>
      <div
        aria-hidden="true"
        class="nhsuk-hint nhsuk-character-count__message nhsuk-character-count__status"
      >
        You have 10 characters remaining
      </div>
      <div
        aria-live="polite"
        class="nhsuk-character-count__sr-status nhsuk-u-visually-hidden"
      >
        You have 10 characters remaining
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the CharacterCount component with max words 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-maxwords="5"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r2:"
        id="textarea_:r2:--label"
      >
        Can you provide more detail?
      </label>
      <div
        class="nhsuk-hint"
        id="textarea_:r2:_:r3:--hint"
      >
        Do not include personal or financial information, for example, your National Insurance number or credit card details.
      </div>
      <textarea
        aria-describedby="textarea_:r2:_:r3:--hint"
        aria-labelledby="textarea_:r2:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="textarea_:r2:"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message nhsuk-u-visually-hidden"
        id="textarea_:r2:-info"
      >
        You have 5 words remaining
      </div>
      <div
        aria-hidden="true"
        class="nhsuk-hint nhsuk-character-count__message nhsuk-character-count__status"
      >
        You have 5 words remaining
      </div>
      <div
        aria-live="polite"
        class="nhsuk-character-count__sr-status nhsuk-u-visually-hidden"
      >
        You have 5 words remaining
      </div>
    </div>
  </div>
</div>
`;
